Core/src/main/java/com/paypal/android/core/analytics/DeviceInspector.kt

```diff
@@ -0,0 +1,63 @@
+package com.paypal.android.core.analytics
+
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import com.paypal.android.core.BuildConfig
+import com.paypal.android.core.analytics.models.DeviceData
+
+class DeviceInspector internal constructor(
+    private val clientSDKVersion: String,
+    private val sdkInt: Int,
+    private val deviceManufacturer: String,
+    private val deviceModel: String,
+    private val deviceProduct: String,
+    private val deviceFingerprint: String
+) {
+
+    constructor() : this(
+        clientSDKVersion = BuildConfig.PAYPAL_SDK_VERSION,
+        sdkInt = Build.VERSION.SDK_INT,
+        deviceManufacturer = Build.MANUFACTURER,
+        deviceModel = Build.MODEL,
+        deviceProduct = Build.PRODUCT,
+        deviceFingerprint = Build.FINGERPRINT
+    )
+
+    fun inspect(context: Context): DeviceData {
+        val packageManager = context.packageManager
+
+        val appName = try {
+            val applicationInfo = packageManager.getApplicationInfo(context.packageName, 0)
+            packageManager.getApplicationLabel(applicationInfo).toString()
+        } catch (ignored: PackageManager.NameNotFoundException) {
+            ""
```

Comment:
need to define default app name for failure scenarios

Reply:
We intentionally wanted the value to be for example `"merchantAppName": ""` in the Analytics JSON. I think this is more explicit when inspecting the metadata than having us inject some string value like `"merchantAppName": "UnknownApp"`. 

We also did this on iOS - what do folks think?

Reply:
We could add something explicit i'm indifferent.

Reply:
I like "N/A." I think it makes sense to have a default value for the rare case where we're unable to determine the merchant's app name.